homeassistant/components/energyzero/diagnostics.py

```diff
@@ -32,30 +30,28 @@ def get_gas_price(data: EnergyZeroData, hours: int) -> float | None:
 
 
 async def async_get_config_entry_diagnostics(
-    hass: HomeAssistant, entry: ConfigEntry
+    hass: HomeAssistant, entry: EnergyZeroConfigEntry
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
-    coordinator: EnergyZeroDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-
     return {
         "entry": {
             "title": entry.title,
         },
         "energy": {
-            "current_hour_price": coordinator.data.energy_today.current_price,
-            "next_hour_price": coordinator.data.energy_today.price_at_time(
-                coordinator.data.energy_today.utcnow() + timedelta(hours=1)
+            "current_hour_price": entry.runtime_data.data.energy_today.current_price,
```

Comment:
I would suggest that you create two intermediate variables:
- `coordinator_data = entry.runtime_data.data` (for gas)
- `energy_today = coordinator_data .energy_today` (for energy)

Then you can simplify the return